TypeCobol/Compiler/CompilationUnit.cs

```diff
@@ -161,29 +161,23 @@ public void RefreshProgramClassDocumentSnapshot()
             lock (lockObjectForProgramClassDocumentSnapshot)
             {
                 // Capture previous snapshot at one point in time
-                CodeElementsDocument codeElementsDocument = CodeElementsDocumentSnapshot;
+                TemporarySemanticDocument temporarySnapshot = TemporaryProgramClassDocumentSnapshot;
 
                 // Check if an update is necessary and compute changes to apply since last version
-                if ((CodeElementsDocumentSnapshot != null) && (ProgramClassDocumentSnapshot == null || ProgramClassDocumentSnapshot.PreviousStepSnapshot.CurrentVersion != codeElementsDocument.CurrentVersion))
+                if ((TemporaryProgramClassDocumentSnapshot != null) && (ProgramClassDocumentSnapshot == null || ProgramClassDocumentSnapshot.PreviousStepSnapshot.CurrentVersion != temporarySnapshot.CurrentVersion))
                 {
-                    // Start perf measurement
-                    var perfStatsForParserInvocation = PerfStatsForProgramClassParser.OnStartRefreshParsingStep();
-
                     // Program and Class parsing is not incremental : the objects are rebuilt each time this method is called
-                    SourceFile root;
-                    List<Diagnostic> newDiagnostics;
-                    Dictionary<CodeElement, Node> nodeCodeElementLinkers = new Dictionary<CodeElement, Node>();
+                    SourceFile root = temporarySnapshot.Root;
+                    List<Diagnostic> diagnostics = temporarySnapshot.Diagnostics ?? new List<Diagnostic>();
```

Comment:
A snapshot only contains diagnostics related to its own phase.
It shall not include diagnostics from previous snapshot.